src/syntax/ast.rs

```diff
@@ -1589,29 +1595,110 @@ impl AstNode for Param {
     }
 }
 
+node! {
+    /// A destructuring pattern: `x` or `(x, _, ..y)`.
+    Pattern
+}
+
+#[derive(Debug, Clone, Hash)]
+pub enum DestructuringKind {
+    /// An identifier: `x`.
+    Ident(Ident),
+    /// An argument sink: `..y`.
+    Sink(Ident),
+}
+
+#[derive(Debug, Clone, Hash)]
+pub enum PatternKind {
+    /// A single identifier: `x`.
+    Ident(Ident),
+    /// A destructuring pattern: `(x, _, ..y)`.
+    Destructure(Vec<DestructuringKind>),
+}
```

Comment:
could you flip the order and add documentation (with an example similar to the others)?

Reply:
I'm not quite sure what examples you mean, I have added some basic documentation for now. Also are the derives here necessary? The code compiles just fine without it.